### ISSUE TITLE
Add Table Formatting

### ISSUE THREAD

**Issue by mariostnr** [SpecFlowOSS/SpecFlow.Rider#31]
In order to quickly write properly structured tables inside of a feature file/scenario
for Developers and Testers
the Rider plugin will transform words into table formats when using pipes " | " characters

whereas currently there is no formatting functionality at all (no document, no tables)

Acceptance criteria:

- strings (text, words) should get automatically transformed into table format when using the pipes " | "character
- the column width should be calculated based on the longest string (text, words)
- strings (text, words) inside of a table column should be aligned left
- "Ctrl K + D" (Visual Studio) should be supported to quickly structure the feature file tables (only the tables, not the whole feature file)
- review the linked bugs and try to avoid them in the Rider plugin
  - https://github.com/SpecFlowOSS/SpecFlow/issues/1593
  - https://github.com/SpecFlowOSS/SpecFlow/issues/1734

**Comment by Socolin** [comment]
I pushed a branch with a first version for formatting.

To configure formatting it's mostly done inside `GherkinFormatterInfoProvider` (Base on `GherkinParser` output).
Some new lines "rules" are configured in `GherkinCodeFormatter.GetMinimalSeparatorByNodeTypes` this one is based on `GherkinLexer` output

If we want to add custom settings, we can add them in `GherkinFormatSettingsKey` then allow the user to configure by adding them to `GherkinFormattingStylePage`

Since table formatting is a more complex than the rules I saw here, I think we may need to write a custom `*Rule`

So far I encounters the following problems:
- I was not able to find a way to add newline between `Examples: ` and the table too.
- I did not find how to add a blank lines between a `GherkinStep` and `GhekinExamplesBlock`
- Each time I reformat there is new lines added after each tables
